File: dawning-admin/src/api/system-log.ts
import axios from 'axios';
import type { IPagedData } from './paged-data';

export interface SystemLog {
  id: string;
  level: string;
  message: string;
  exception?: string;
  stackTrace?: string;
  source?: string;
  userId?: string;
  username?: string;
  ipAddress?: string;
  userAgent?: string;
  requestPath?: string;
  requestMethod?: string;
  statusCode?: number;
  createdAt: string;
}

export interface SystemLogQueryParams {
  level?: string;
  keyword?: string;
  userId?: string;
  username?: string;
  ipAddress?: string;
  requestPath?: string;
  startDate?: string;
  endDate?: string;
  page?: number;
  pageSize?: number;
}

export interface CreateSystemLogDto {
  level: string;
  message: string;
  exception?: string;
  stackTrace?: string;
  source?: string;
  userId?: string;
  username?: string;
  ipAddress?: string;
  userAgent?: string;
  requestPath?: string;
  requestMethod?: string;
  statusCode?: number;
}

/**
 * 分页获取系统日志列表
 */
export function getSystemLogList(params: SystemLogQueryParams) {
  return axios.get<IPagedData<SystemLog>>('/api/systemlog/paged', {
    params,
  });
}

/**
 * 根据ID获取系统日志详情
 */
export function getSystemLog(id: string) {
  return axios.get<SystemLog>(`/api/systemlog/${id}`);
}

/**
 * 删除指定日期之前的日志
 */
export function cleanupSystemLogs(beforeDate: string) {
  return axios.delete('/api/systemlog/cleanup', {
    params: { beforeDate },
  });
}

/**
 * 创建测试日志
 */
export function createTestLog(data: CreateSystemLogDto) {
  return axios.post('/api/systemlog/test', data);
}

File: apps/gateway/src/Dawning.Identity.Domain/Aggregates/Administration/SystemLog.cs
using System;
using Dawning.Identity.Domain.Core.Interfaces;

namespace Dawning.Identity.Domain.Aggregates.Administration
{
    /// <summary>
    /// System log aggregate root
    /// Used for recording application logs and exception information
    /// </summary>
    public class SystemLog : IAggregateRoot
    {
        /// <summary>
        /// Log unique identifier
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// Log level (Info, Warning, Error)
        /// </summary>
        public string Level { get; set; } = string.Empty;

        /// <summary>
        /// Log message
        /// </summary>
        public string Message { get; set; } = string.Empty;

        /// <summary>
        /// Exception information (exception type and message)
        /// </summary>
        public string? Exception { get; set; }

        /// <summary>
        /// Exception stack trace
        /// </summary>
        public string? StackTrace { get; set; }

        /// <summary>
        /// Exception source
        /// </summary>
        public string? Source { get; set; }

        /// <summary>
        /// Operating user ID
        /// </summary>
        public Guid? UserId { get; set; }

        /// <summary>
        /// Operating username
        /// </summary>
        public string? Username { get; set; }

        /// <summary>
        /// IP address
        /// </summary>
        public string? IpAddress { get; set; }

        /// <summary>
        /// User agent
        /// </summary>
        public string? UserAgent { get; set; }

        /// <summary>
        /// Request path
        /// </summary>
        public string? RequestPath { get; set; }

        /// <summary>
        /// Request method (GET, POST, PUT, DELETE, etc.)
        /// </summary>
        public string? RequestMethod { get; set; }

        /// <summary>
        /// HTTP status code
        /// </summary>
        public int? StatusCode { get; set; }

        /// <summary>
        /// Created time
        /// </summary>
        public DateTime CreatedAt { get; set; }

        /// <summary>
        /// Timestamp (used for sorting and query optimization)
        /// </summary>
        public long Timestamp { get; set; }
    }
}
